test-data/unit/check-modules.test

```diff
@@ -1416,3 +1416,40 @@ reveal_type(f())  # E: Revealed type is 'types.ModuleType'
 reveal_type(types)  # E: Revealed type is 'types.ModuleType'
 
 [builtins fixtures/module.pyi]
+
+[case testModuleAssignment]
+import m
+m2 = m
+reveal_type(m2.a)  # E: Revealed type is 'builtins.str'
+
+[file m.py]
+a = 'foo'
+
+[builtins fixtures/module.pyi]
+
+[case testChainedModuleAssignment]
+import m
+m3 = m2 = m
+reveal_type(m2.a)  # E: Revealed type is 'builtins.str'
+reveal_type(m3.a)  # E: Revealed type is 'builtins.str'
+
+[file m.py]
+a = 'foo'
+
+[builtins fixtures/module.pyi]
+
+[case testMultiModuleAssignment]
+import m, n
+m2, n2, (m3, n3) = m, n, [m, n]
+reveal_type(m2.a)  # E: Revealed type is 'builtins.str'
+reveal_type(n2.b)  # E: Revealed type is 'builtins.str'
+reveal_type(m3.a)  # E: Revealed type is 'builtins.str'
+reveal_type(n3.b)  # E: Revealed type is 'builtins.str'
+
+[file m.py]
+a = 'foo'
+
+[file n.py]
+b = 'bar'
+
+[builtins fixtures/module.pyi]
```

Comment:
I would like to see few more tests. For example, a test that shows this failing:
```python
import m
x, y = m  # must be an error here (Module object is not iterable or similar)
```
and in general more failures, like failure on attempted access/assignment to a non-existing module attribute.

Also probably some chained assignments:
```python
import m
x = m
y = x
reveal_type(y.a) # 'builtins.str'
```